www/main/navigator/shabad/ShabadHeader.jsx

```diff
@@ -46,11 +52,37 @@ const ShabadHeader = () => {
 
   return (
     <div className="shabad-pane-header">
+      {activeShabadId && !isLoading && userToken && (
+        <button
+          className={classNames('button fav-btn', favShabadIndex >= 0 && 'unfav-btn')}
+          ref={favBtnRef}
+          title={i18n.t('SHABAD_PANE.FAV_BTN_TOOLTIP')}
+          onClick={() => {
+            if (favShabadIndex < 0) {
+              addToFav(activeShabadId, activeVerseId, userToken);
+            } else {
+              favShabad.splice(favShabadIndex, 1);
+              removeFromFav(activeShabadId, userToken);
+              setFavShabad([...favShabad]);
+            }
+            const fetchProgress = fetchFavShabad(userToken);
+            setLoading(true);
+            fetchProgress.then((data) => {
+              setFavShabad([...data]);
+              setLoading(false);
+            });
+          }}
```

Comment:
function definition should be outside the jsx for longer functions like this

Reply:
Fixed